xarray/core/nputils.py

```diff
@@ -232,10 +232,17 @@ def _nanpolyfit_1d(arr, x, rcond=None):
     out = np.full((x.shape[1] + 1,), np.nan)
     mask = np.isnan(arr)
     if not np.all(mask):
-        out[:-1], out[-1], _, _ = np.linalg.lstsq(x[~mask, :], arr[~mask], rcond=rcond)
+        out[:-1], resid, rank, _ = np.linalg.lstsq(x[~mask, :], arr[~mask], rcond=rcond)
+        out[-1] = resid or np.nan
```

Comment:
```suggestion
        out[-1] = resid if resid.size > 0 else np.nan
```



This raises a DeprecationWarning:
```
/home/vsts/work/1/s/xarray/core/nputils.py:236: DeprecationWarning:
The truth value of an empty array is ambiguous. Returning False, but in future this will result in an error.
Use `array.size > 0` to check that an array is not empty.
    out[-1] = resid or np.nan
```